### PR TITLE
support carton in every environment

### PR DESCRIPTION
**Pull request by rishi-raj-jain**
This PR aims at extending carton in every environment: npm, pnpm, bun and yarn.

### CODE REVIEW

**Comment by lilnasy** [comment]
Thanks for getting this started! This is definitely the next step. 

I would be surprised if the current approach works, however, since "dist" is not in `astro.packageJson.exports`.

**Comment by lilnasy** [comment]
Yeah, the approach doesn't work.

image

(The way you have to test it is by creating a tarball with `pnpm pack`.)

No worries, though, I already know a way it could work.

**Comment by rishi-raj-jain** [comment]
Thanks for the quick review!

> No worries, though, I already know a way it could work.

May you help me with that? I'd love to contribute and learn!

**Comment by rishi-raj-jain** [comment]
woah, I didn't know `import.meta.resolve('module')` was an API

**Comment by lilnasy** [comment]
Just needs a changeset and readme update now.

**Comment by rishi-raj-jain** [comment]
Sure, on it

Curious, does `import.meta.resolve('node:crypto')` work in similar fashion? what if `node:crypto` is not available?

**Comment by rishi-raj-jain** [comment]
@lilnasy 

Updated the README and added changeset

**Comment by lilnasy** [comment]
> Curious, does import.meta.resolve('node:crypto') work in similar fashion? what if node:crypto is not available?

Just tried it and it just returns `"node:crypto"` unchanged.

File: .changeset/curvy-buttons-fry.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"astro-carton": minor
+---
+
+support carton in every environment
```
Comment (lilnasy) [inline .changeset/curvy-buttons-fry.md:5]:
```suggestion
Adds compatibility with all package managers.
```

**Review by lilnasy — APPROVED** [review]
Thanks for taking this on!

**Comment by rishi-raj-jain** [comment]
My pleasure!

**Pull request merged**